sympy/printing/ccode.py

```diff
@@ -336,20 +336,21 @@ def ccode(expr, assign_to=None, **settings):
     tau = x;
     }
 
-    Support for loops is provided through ``Indexed`` types. With
-    ``contract=True`` these expressions will be turned into loops, whereas
-    ``contract=False`` will just print the assignment expression that should be
-    looped over:
+    Support for loops is provided through ``Indexed`` types; automatic summation
+    of repeated indices occurs when expressions are wrapped in ``EinsteinSum``.
 
     >>> from sympy import Eq, IndexedBase, Idx
+    >>> from __future__ import print_function
     >>> len_y = 5
     >>> y = IndexedBase('y', shape=(len_y,))
     >>> t = IndexedBase('t', shape=(len_y,))
     >>> Dy = IndexedBase('Dy', shape=(len_y-1,))
     >>> i = Idx('i', len_y-1)
     >>> e=Eq(Dy[i], (y[i+1]-y[i])/(t[i+1]-t[i]))
-    >>> ccode(e.rhs, assign_to=e.lhs, contract=False)
-    'Dy[i] = (y[i + 1] - y[i])/(t[i + 1] - t[i]);'
+    >>> print(ccode(e.rhs, assign_to=e.lhs))
+    for (int i=0; i<4; i++){
+       Dy[i] = (y[i + 1] - y[i])/(t[i + 1] - t[i]);
+    }
 
```

Comment:
How should one call ccode to generate the original output:

    Dy[i] = (y[i + 1] - y[i])/(t[i + 1] - t[i]);

I thought not using EinsteinSum would do this?